src/EditorFeatures/Core/Implementation/AutomaticCompletion/BraceCompletionSessionProvider.BraceCompletionSession.cs

```diff
@@ -256,7 +278,13 @@ public void PostReturn()
 
                     if (closingSnapshotPoint.Position > 0 && HasNoForwardTyping(this.GetCaretPosition().Value, closingSnapshotPoint.Subtract(1)))
                     {
-                        var changesAfterReturn = _session.GetChangesAfterReturn(this, CancellationToken.None);
+                        var context = GetBraceCompletionContext();
+                        if (context == null)
+                        {
+                            return;
+                        }
+
+                        var changesAfterReturn = _service.GetTextChangeAfterReturnAsync(context, CancellationToken.None).WaitAndGetResult(CancellationToken.None);
```

Comment:
:'(  with all the CT.Nones.  We might want to file feedback with editor that these are potentially expensive and CT should be hooked up.